units/pod_initializing.go

```diff
@@ -102,4 +108,8 @@ func (j *podInitJob) Run(ctx context.Context) {
 			"job":     j.ID(),
 		})
 	}
+
+	if err := j.pod.UpdateStatus(pod.StatusStarting); err != nil {
```

Comment:
You have to set the resources on the pod to reflect those that you created in `CreatePod` before you can update the state. You can get them from `ECSPod.Info(ctx)`.